var/spack/repos/builtin/packages/astyle/package.py

```diff
@@ -25,28 +25,24 @@
 from spack import *
 
 
-class Astyle(Package):
+class Astyle(EditableMakefile):
     """
     A Free, Fast, and Small Automatic Formatter for C, C++, C++/CLI,
     Objective-C, C#, and Java Source Code.
     """
     homepage = "http://astyle.sourceforge.net/"
-    url      = "http://downloads.sourceforge.net/project/astyle/astyle/astyle%202.04/astyle_2.04_linux.tar.gz"
+    url = "http://downloads.sourceforge.net/project/astyle/astyle/astyle%202.04/astyle_2.04_linux.tar.gz"
 
     version('2.04', '30b1193a758b0909d06e7ee8dd9627f6')
 
-    def install(self, spec, prefix):
+    parallel = False
 
-        with working_dir('src'):
-            # we need to edit the makefile in place to set compiler:
-            make_file = join_path(self.stage.source_path,
-                                  'build', 'gcc', 'Makefile')
-            filter_file(r'^CXX\s*=.*', 'CXX=%s' % spack_cxx, make_file)
+    def wdir(self):
+        return join_path(self.stage.source_path, 'build', self.compiler.name)
 
-            make('-f',
-                 make_file,
-                 parallel=False)
+    def edit(self, spec, prefix):
+        makefile = join_path(self.wdir(), 'Makefile')
+        filter_file(r'^CXX\s*=.*', 'CXX=%s' % spack_cxx, makefile)
 
-            mkdirp(self.prefix.bin)
-            install(join_path(self.stage.source_path, 'src', 'bin', 'astyle'),
```

Comment:
i think this is not correct. `astyle` does not have `install` target, so here it was a manual installation of a single binary `astyle` into `self.prefix.bin`, created above.